app/controllers/desks_controller.rb

```diff
@@ -45,6 +45,11 @@ def broadcast_needed?
   end
 
   def broadcast_desk_update
-    ActionCable.server.broadcast('desks', @desk.slice('id', 'name', 'occupied'))
+
+    ActionCable.server.broadcast('desks',
+                                 desk_info: @desk.slice('id', 'name', 'occupied'),
+                                 element_style: helpers.desk_class(@desk),
+                                 element_text: helpers.desk_status(@desk),
+                                 element_id: helpers.desk_element_id(@desk))
```

Comment:
@DominikSchmidtlein @erickadbay 

How do you like them apples? (but really though I need feedback, Im not sure if this is the best way to do this in rails)

I changed it so that we broadcast the element_style, text and id using the helper methods.

This does two major things:
- Cleans up javascript code in desks.js
- Allows us to use our helper methods to determine the style / text/ id (de-couples javascript logic from our controller logic - single responsibility principle)

Reply:
In this context, single responsibility means that the javascript should only be responsible for receiving the data and updating the view. It should NOT be responsible for determining how the view should be changed - this Is already handled by our helper methods

Reply:
I think it's very fair for javascript to determine classes and such since it's client side logic. I'd leave that logic out of the controller since the controller shouldn't care how you display what it's sending to the view.

Reply:
If we were to ever change the class of the buttons, we'd have to change it in two places - javascript and the view. When we delegate this to the helper methods, it keeps all changes in line.

Reply:
Plus - there aren't any tests for our javascript code, so if we were to ever change the helpers, there would be nothing to catch the lack of changes in javascript

Reply:
Still, I don't believe it's something you should broadcast from within the controller. The controller should be completely agnostic on how to display things.

There are ways of test the UI behaviour that you want, through integration tests (i.e test that the button class changes to `btn-outline-danger` after updating the desk), although i don't think it's necessary since the behaviour will likely change in the future.

Reply:
An issue with this approach is that it couples the status update to a UI change. The backend should work with different front ends regardless of how they choose to display changes.

Reply:
@DominikSchmidtlein @erickadbay do you both agree that I should revert this specific change so that we don't have a coupled view?

Reply:
Yes, I agree.

Reply:
👍  to revert the change